### PR TITLE
private/xueqingz/CP-32724

### PR DESCRIPTION
**Pull request by xueqingz**
XenRT test pass 2811723,2811724,2811725,2811726,2811728,2811732.
new XenRT test pass : 2817461,2817462,2817463,2817464,2817465,2817466,2817467.

### CODE REVIEW

File: transfervm/overlay/sbin/hotplug.stunnel [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 sslversion=$(/usr/bin/xenstore-read "vm-data/transfer/ssl_version")
 if [ "$sslversion" == "" ]; then
-    sslversion="all"
+    sslversion="TLSv1.2"
```
Comment (edwintorok) [inline transfervm/overlay/sbin/hotplug.stunnel]: can you unconditionally set this to TLSv1.2? We are removing older protocols in another REQ.

Comment (edwintorok) [reply transfervm/overlay/sbin/hotplug.stunnel]: Same below about sslciphers, it should only use TLSv1.2

File: transfervm/overlay/sbin/hotplug.stunnel [hunk 2]
```diff
@@ -2,16 +2,13 @@
...
 fi
 
 sslciphers=$(/usr/bin/xenstore-read "vm-data/transfer/ssl_ciphers")
 if [ "$sslciphers" == "" ]; then
     if [ "$sslversion" == "TLSv1.2" ]; then
-        sslciphers="!SSLv2:RSA+AES128-SHA256"
-    else
-        sslciphers="!SSLv2:RSA+AES128-SHA256:RSA+AES256-SHA:RSA+AES128-SHA:RSA+RC4-SHA:RSA+RC4-MD5:RSA+DES-CBC3-SHA"
-    fi
+        sslciphers="!SSLv2:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-GCM-SHA384:AES256-SHA256:AES128-SHA256"
```
Comment (xueqingz) [inline transfervm/overlay/sbin/hotplug.stunnel]: change to the new cipher list

**Review by Zhengchai — APPROVED** [review]
Looks good to me. Approve.

File: transfervm/overlay/sbin/hotplug.stunnel (hunk 1 above)
Comment (minli1) [reply transfervm/overlay/sbin/hotplug.stunnel]: I think here still remains one comment from Edwin. We can set sslversion and sslciphers to TLSv1.2 unconditionally.

**Review by minglumlu — APPROVED** [review]

**Comment by edwintorok** [comment]
Regarding sslversion in xenstore: we're removing the configurability of SSL version from elsewhere in the product, so it would make sense to remove it from here also. Hopefully by the time we implement TLS1.3 support we would have replaced TransferVM with something else (e.g. done directly by the toolstack).
Also when we update to CentOS 8 there are system-wide crypto policy configurations that would make these kinds of choices a lot easier, instead of having to make them in each component.

**Comment by edwintorok** [comment]
This shows up in our list of open PRs, is there anything more to be done here, or can this be merged? (I don't have permission to merge it)

**Comment by Zhengchai** [comment]
Merged!

**Pull request merged**